Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp

```diff
@@ -324,84 +324,72 @@ auto MediaStreamTrack::takePhoto(PhotoSettings&& settings) -> Ref<TakePhotoPromi
     if (m_ended)
         return TakePhotoPromise::createAndReject(Exception { ExceptionCode::InvalidStateError, "Track has ended"_s });
 
-    TakePhotoPromise::Producer producer;
-    Ref<TakePhotoPromise> promise = producer;
-
-    m_private->takePhoto(WTFMove(settings))->whenSettled(RunLoop::main(),
-        [protectedThis = Ref { *this }, producer = WTFMove(producer)] (auto&& result) mutable {
+    return m_private->takePhoto(WTFMove(settings))->whenSettled(RunLoop::main(),
+        [protectedThis = Ref { *this }] (auto&& result) mutable -> Ref<TakePhotoPromise> {
```

Comment:
the compiler can't deduct this one ? `-> Ref<TakePhotoPromise>` shouldn't be needed

Reply:
Changed